velox/type/Date.cpp

```diff
@@ -27,15 +27,16 @@ std::string Date::toString() const {
   // Find the number of seconds for the days_;
   // Casting 86400 to int64 to handle overflows gracefully.
   int64_t daySeconds = days_ * (int64_t)(86400);
-  auto tmValue = gmtime((const time_t*)&daySeconds);
-  if (!tmValue) {
-    VELOX_FAIL("Can't convert days to dates: {}", days_);
-  }
+  std::tm tmValue;
+  VELOX_USER_CHECK_NOT_NULL(
+      gmtime_r((const time_t*)&daySeconds, &tmValue),
+      "Can't convert days to dates: {}",
+      days_);
```

Comment:
`time_t` is `long` instead of `long long`, and `int64_t` is `long long`.  

Casting `int64_t*` to `time_t*` (i.e. converting `long long*` to `long*`) is a bit obscure and non-portable (although, I know, we don't currently run on 32-bit machines).

A better way to write it might be:
```
  // Casting 'days_' to time_t to avoid overflow in multiplication.
  const time_t daySeconds = static_cast<time_t>(days_) * 86400L;
  std::tm tmValue;
  VELOX_USER_CHECK_NOT_NULL(
      gmtime_r(&daySeconds, &tmValue),    //  <----  Casting is no longer required.
      "Can't convert days to dates: {}",
      days_);
```
what do you think?

Reply:
Yes, it's better. Thanks. Could you help review it @laithsakka @aditi-pandit